File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    r,cache=0,[]
    if cacheSize==0:
        r=len(cities)*5
    else:
        for city in cities:
            city=city.lower()
            if city in cache:
                cache.remove(city)
                cache+=[city]
                r+=1
            else: # if city not in cache
                if len(cache)>=cacheSize:
                    cache.pop(0)
                cache+=[city]
                r+=5
    return r

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int cacheSize, vector<string> cities) {
    int r=0;
    vector<string> cache;
    if(cacheSize==0)
        r=cities.size()*5;
    else {
        for(int i=0;i<cities.size();i++) {
            transform(cities[i].begin(),cities[i].end(),cities[i].begin(),[](auto c) {return tolower(c);});
            if(find(cache.begin(),cache.end(),cities[i])!=cache.end()) {
                cache.erase(find(cache.begin(),cache.end(),cities[i]));
                cache.push_back(cities[i]);
                r+=1;
            } else {
                if(cache.size()>=cacheSize)
                    cache.erase(cache.begin());
                cache.push_back(cities[i]);
                r+=5;
            }
        }
    }
    return r;
}